atc/exec/task_step.go

```diff
@@ -206,6 +210,20 @@ func (step *TaskStep) run(ctx context.Context, state RunState) error {
 
 	step.delegate.Initializing(logger)
 
+	if step.policyChecker != nil && config.ImageResource != nil {
```

Comment:
This could be circumvented by using [`image:`](https://concourse-ci.org/jobs.html#schema.step.task-step.image) and pointing to an image fetched by a `get` step or provided by a task output. Or by using a resource type that doesn't use `repository:` in its configuration.

Reply:
Removed the check from task step.